src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -165,6 +166,12 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 register_cli_argument('network application-gateway create', 'cert_password', help='The certificate password', arg_group='Gateway')
 register_cli_argument('network application-gateway create', 'http_settings_port', help='The HTTP settings port.', arg_group='Gateway')
 
+for item in ['create', 'http-settings']:
+    if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-12-01'):
+        register_cli_argument('network application-gateway {}'.format(item), 'connection_draining_timeout', type=int, help='The time in seconds after a backend server is removed during which on open connection remains active. Range: 0 (disabled) to 3600', arg_group='Gateway' if item == 'create' else None)
+    else:
+        register_cli_argument('network application-gateway {}'.format(item), 'connection_draining_timeout', ignore_type)
```

Comment:
Do you need the ignore type (and the else statement in general)?
If `connection_draining_timeout` is not a param in the other API version, the param just won't be shown.